don't need to compress the binary size in Mac build

## Bug Report

### What version of TiKV are you using?
Master

### What operating system and CPU are you using?

MacOSX

### Steps to reproduce

```
make dist_unportable_release
```

### What did you expect?

make passed

### What did happened?

```
# Reduce binary size by compressing binaries.
# FIXME: Currently errors with `Couldn't find DIE referenced by DW_AT_abstract_origin`
# dwz /Users/tangliu/program/tikv/target/release/tikv-server
# FIXME: https://sourceware.org/bugzilla/show_bug.cgi?id=24764
# dwz /Users/tangliu/program/tikv/target/release/tikv-ctl
objcopy --compress-debug-sections=zlib-gnu /Users/tangliu/program/tikv/target/release/tikv-server
make[2]: objcopy: No such file or directory
make[2]: *** [build_dist_release] Error 1
make[1]: *** [dist_release] Error 2
make: *** [dist_unportable_release] Error 2
```

I think we don't need to do the compression on MacOSX.

--- a/Makefile
+++ b/Makefile
@@ -157,6 +157,7 @@ dist_release:
 # additional sanity checks and file movement.
 build_dist_release:
 	make x-build-dist
+ifeq ($(shell uname),Linux) # Macs don't have objcopy
 	# Reduce binary size by compressing binaries.
 	# FIXME: Currently errors with `Couldn't find DIE referenced by DW_AT_abstract_origin`
 	# dwz ${CARGO_TARGET_DIR}/release/tikv-server
@@ -164,6 +165,7 @@ build_dist_release:
 	# dwz ${CARGO_TARGET_DIR}/release/tikv-ctl
 	objcopy --compress-debug-sections=zlib-gnu ${CARGO_TARGET_DIR}/release/tikv-server
 	objcopy --compress-debug-sections=zlib-gnu ${CARGO_TARGET_DIR}/release/tikv-ctl
+endif
 
 # Distributable bins with SSE4.2 optimizations
 dist_unportable_release:

